### PR TITLE
Have CI do a fmt as well

### PR DESCRIPTION
**Pull request by Hoverbear**
Currently we're not using `rustfmt` via CI for Raft, and we should be! This configures it.

### CODE REVIEW

**Comment by Hoverbear** [comment]
This is failing due to the Rust version being too old to have the `rustfmt-preview` component. I'll do this in another PR.

**Comment by Hoverbear** [comment]
Based this off #56 so it will be green. Please merge #56 first though.

File: .travis.yml [hunk 1]
```diff
@@ -3,21 +3,26 @@ sudo: false
...
+  # This build uses stable and checks rustfmt (clippy is not on stable).
   - rust: stable
-  - rust: nightly-2018-01-12
+    install:
+      - rustup component add rustfmt-preview
+  # This build uses the nightly used by TiKV and checks both rustfmt & clippy.
+  - rust: nightly-2018-04-06
     install:
       - export PATH="$PATH:$HOME/.cargo/bin"
+      - rustup component add rustfmt-preview
```
Comment (BusyJay) [inline .travis.yml]: What if the format is different from stable channel?

Comment (Hoverbear) [reply .travis.yml]: :+1: Great catch, you're totally right. We should just run `rustfmt` on `stable`.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**